packages/video_player/video_player/lib/video_player.dart

```diff
@@ -537,6 +549,29 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
     );
   }
 
+  Future<void> _applyMaxVideoResolution() async {
+    if (_isDisposedOrNotInitialized) {
+      return;
+    }
+
+    // Setting the playback speed on iOS will trigger the video to play. We
+    // prevent this from happening by not applying the playback speed until
+    // the video is manually played from Flutter.
+    if (!value.isPlaying) {
```

Comment:
Is this check necessary for this method? It looks like it was copied from `_applyPlaybackSpeed()`.